### PR TITLE
Describe get path params and get node params better

### PR DESCRIPTION
**Pull request by pekura**
- describe get path params and get node params in Luigi Client doc better
- fix broken links to Luigi Client doc in the main readme

### CODE REVIEW

File: client/luigi-client.js [hunk 1]
```diff
@@ -144,16 +144,19 @@
       return currentContext.context;
     },
     /**
-     * Returns the configuration object of the active navigation Node.
-     * @returns {Object} node parameters.
+     * Returns the node parameters of the active URL.
```
[the remark this answers is not in the document]
Comment (pekura) [reply client/luigi-client.js]: So, not capitalized.

File: docs/luigi-client-api.md [hunk 2]
```diff
@@ -30,15 +30,18 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** node parameters where object property name is the node parameter name (without the prefix) and its value is the value of the node parameter (e.g. {sort: 'asc', page: 3})
 
 ### getPathParams
 
 Returns the dynamic path parameters of the active URL.
+Path parameters are defined by navigation nodes with dynamic pathSegment (starting with ':', e.g. ':productId').
+All such parameters in the current navigation path (as defined by the active URL) are returned.
 
-Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** path parameters.
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** path parameters where object property name is the path parameter name and its value is the actual value of the path parameter (e.g. {productId: '1234', ...})
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: done

File: docs/luigi-client-api.md [hunk 3]
```diff
@@ -30,15 +30,18 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+Returns the node parameters of the active URL.
+Node parameters are defined like URL query parameters but with a specific prefix (default is '~') that marks them to be passed to the micro-frontend view (e.g. <https://my.luigi.app/home/products?~sort=asc~page=3>).
 
-Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** node parameters.
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** node parameters where object property name is the node parameter name (without the prefix) and its value is the value of the node parameter (e.g. {sort: 'asc', page: 3})
 
 ### getPathParams
 
 Returns the dynamic path parameters of the active URL.
+Path parameters are defined by navigation nodes with dynamic pathSegment (starting with ':', e.g. ':productId').
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: done

File: docs/luigi-client-api.md [hunk 4]
```diff
@@ -30,15 +30,18 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
 
 ### getNodeParams
 
-Returns the configuration object of the active navigation Node.
+Returns the node parameters of the active URL.
+Node parameters are defined like URL query parameters but with a specific prefix (default is '~') that marks them to be passed to the micro-frontend view (e.g. <https://my.luigi.app/home/products?~sort=asc~page=3>).
 
-Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** node parameters.
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** node parameters where object property name is the node parameter name (without the prefix) and its value is the value of the node parameter (e.g. {sort: 'asc', page: 3})
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: done

File: docs/luigi-client-api.md [hunk 5]
```diff
@@ -30,15 +30,18 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
 
 ### getNodeParams
 
-Returns the configuration object of the active navigation Node.
+Returns the node parameters of the active URL.
+Node parameters are defined like URL query parameters but with a specific prefix (default is '~') that marks them to be passed to the micro-frontend view (e.g. <https://my.luigi.app/home/products?~sort=asc~page=3>).
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: done

File: README.md [hunk 6]
```diff
@@ -24,9 +24,9 @@ Luigi core is the part of Luigi that enables navigation, security and routing. T
 
 Luigi client enables integration of Luigi in views. You can easily configure the lifecycle and links of the client, as well as customize the appearance of your navigation.
 
-- [Lifecycle](docs/lifecycle.md) gives you details on the life cycle of listeners, navigation Nodes and Event data.
-- [Link Manager](docs/link-manager.md) shows how to configure the Link Manager.
-- [UX Manager](docs/ux-manager.md) shows how to set up the UX Manager.
+- [Lifecycle](docs/luigi-client-api.md#lifecycle) gives you details on the life cycle of listeners, navigation Nodes and Event data.
+- [Link Manager](docs/luigi-client-api.md#linkmanager) shows how to configure the Link Manager.
```
[the remark this answers is not in the document]
Comment (pekura) [reply README.md]: done

File: docs/luigi-client-api.md [hunk 7]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+// Can be chained with context settings functions like this:
+LuigiClient.linkManager.fromContext("currentTeam").withParams({foo: "bar"}).navigate("path")
+```
+
+Returns **[linkManager](#linkmanager)** link manager instance.
+
+### hasBack
+
+Checks if there are one or more preserved views. Can be used to show a back button.
```
Comment (bszwarc) [inline docs/luigi-client-api.md]: ` Can be used to show a **back** button` I would bold **back** as a UI part (as it most likely will be a UI part).

File: docs/luigi-client-api.md [hunk 8]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+Returns **[boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** a boolean with the information if there is a preserved view available to which a user can return.
```
Comment (bszwarc) [inline docs/luigi-client-api.md]:
```
Returns the **[boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean)**  indicating if there is a preserved view available to which the user can return. 

```
the second mentioning of **boolean** is redundant.

File: docs/luigi-client-api.md [hunk 9]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+### addBackdrop
+
+Adds a backdrop to block the top and side navigation. It is based on Fundamental UI Modal, which you can use in your micro front-end to achieve the same behavior.
+
+### removeBackdrop
+
+Removes the backdrop.
+
+# Node parameters are defined like URL query parameters but with a specific prefix allowing them to be passed to the micro front-end view.  The default prefix is **~** and it is used in the following example: `https://my.luigi.app/home/products?~sort=asc~page=3`.
```
Comment (bszwarc) [inline docs/luigi-client-api.md]: Something went wrong here, I guess an unresolved merging conflict - below we can find the same document generated for the second time.

File: docs/luigi-client-api.md [hunk 10]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
 
 ### getNodeParams
 
+Returns the node parameters of the active URL.
+Node parameters are defined like URL query parameters but with a specific prefix allowing them to be passed to the micro front-end view.  The default prefix is **~** and it is used in the following example: `https://my.luigi.app/home/products?~sort=asc~page=3`.
+
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** node parameters, where the object property name is the node parameter name without the prefix, and its value is the value of the node parameter. For example `{sort: 'asc', page: 3}`
```
Comment (bszwarc) [inline docs/luigi-client-api.md]: Dot missing at the end.

File: docs/luigi-client-api.md [hunk 11]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+### getPathParams
+
+Returns the dynamic path parameters of the active URL.
+Path parameters are defined by navigation nodes with a dynamic **pathSegment** value starting with **:** such as `productId`.
+All such parameters in the current navigation path (as defined by the active URL) are returned.
+
+Returns **[Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object)** path parameters, where the object property name is the path parameter name without the prefix, and its value is the actual value of the path parameter. For example `{productId: 1234, ...}`
```
Comment (bszwarc) [inline docs/luigi-client-api.md]: Dot missing at the end.

File: docs/README.md [hunk 12]
```diff
@@ -13,6 +13,6 @@ This directory contains the following documents that relate to the project:
 
 **Luigi Client**
 
-* [Lifecycle](luigi-client-api.md#lifecycle) gives you details on the life cycle of listeners, navigation nodes, and Event data.
-* [Link Manager](luigi-client-api.md#linkmanager) guides you through the configuration of the Link Manager.
-* [UX Manager](luigi-client-api.md#uxmanager) shows you how to set up the UX Manager.
\ No newline at end of file
+* [Lifecycle](luigi-client-api.md#lifecycle) gives you details on the life cycle of listeners, navigation nodes and event data.
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/README.md]: It is a Luigi event, not kubernetes-related, so we keep it non-capitalized.

File: docs/luigi-client-api.md (hunk 10 above)
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md (hunk 11 above)
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md [hunk 13]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+Lets you navigate to another route. Use the Link Manager instead of an internal router to:
+
+-   route inside micro front-ends
+-   reflect the route
+-   keep the navigation state in Luigi
+
+### navigate
+
+Navigates to the given path in the hosting Luigi app. Contains either a full absolute path or a relative path without a leading slash that uses the active route as a base. This is a classical navigation.
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md [hunk 14]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+#### Parameters
+
+-   `path` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)** path to be navigated to
+-   `sessionId` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)** current Luigi sessionId
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md [hunk 15]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+```javascript
+LuigiClient.linkManager().navigate('/overview')
+LuigiClient.linkManager().navigate('users/groups/stakeholders')
+LuigiClient.linkManager().navigate('/settings', null, true) // preserve view
+```
+
+### fromContext
+
+Sets the current navigation context to that of a specific parent node that has the [navigationContext](navigation-configuration.md) field declared in its navigation configuration. This navigation context is then used by navigate function.
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md [hunk 16]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+```javascript
+LuigiClient.linkManager().fromContext('project').navigate('/settings')
+```
+
+Returns **[linkManager](#linkmanager)** link manager instance.
+
+### fromClosestContext
+
+Sets the current navigation context, which is then used by the navigate function. This has to be a parent navigation context, it is not possible to go to child navigation contexts.
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md [hunk 17]
```diff
@@ -30,15 +30,148 @@ Returns **{idToken: [string](https://developer.mozilla.org/docs/Web/JavaScript/R
...
+
+```javascript
+LuigiClient.linkManager().fromClosestContext().navigate('/users/groups/stakeholders')
+```
+
+Returns **[linkManager](#linkmanager)** link manager instance.
+
+### withParams
+
+Sends node parameters to the route, which are then used by the navigate function. Use it optionally in combination with any of the navigation functions and receive it as part of the context object in Luigi Client.
```
[the remark this answers is not in the document]
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md (hunk 7 above)
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md (hunk 8 above)
Comment (pekura) [reply docs/luigi-client-api.md]: Done.

File: docs/luigi-client-api.md (hunk 9 above)
Comment (pekura) [reply docs/luigi-client-api.md]: Fixed.

File: docs/luigi-client-api.md [hunk 18]
```diff
@@ -107,17 +110,39 @@ Sends node parameters to the route, which are then used by the navigate function
...
+let pathExists;
+ this.luigiClient
+ .linkManager()
+ .pathExists('projects/pr2')
+ .then(
+   (pathExists) => {  }
+ );
+```
+
+Returns **[promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise)** A promise that will resolve to a Boolean variable specifying if the path exists or not.
```
Comment (bszwarc) [inline docs/luigi-client-api.md]: `A promise which resolves to a Boolean variable specifying if the path exists or not.`

File: docs/luigi-client-api.md [hunk 19]
```diff
@@ -107,17 +110,39 @@ Sends node parameters to the route, which are then used by the navigate function
...
-// Can be chained with context settings functions like this: 
+// Can be chained with context settings functions like this:
 LuigiClient.linkManager.fromContext("currentTeam").withParams({foo: "bar"}).navigate("path")
 ```
 
 Returns **[linkManager](#linkmanager)** link manager instance.
 
+### pathExists
+
+Checks if a path exists in the main application, i.e., if that path can be navigated to. This helper method can be used e.g. to conditionally display a DOM element like a button.
```
Comment (bszwarc) [inline docs/luigi-client-api.md]: `Checks if a path you can navigate to exists in the main application. You can use this helper method to perform actions such as conditional display of a DOM element like a button. `

File: docs/luigi-client-api.md [hunk 20]
```diff
@@ -107,17 +110,39 @@ Sends node parameters to the route, which are then used by the navigate function
...
+
+Returns **[promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise)** A promise that will resolve to a Boolean variable specifying if the path exists or not.
+
 ### hasBack
 
-Checks if there are one or more preserved views. Can be used to show a back button.
+Checks if there are one or more preserved views. Can be used to show a **back** button.
 
-Returns **[boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** a boolean with the information if there is a preserved view available to which a user can return.
+Returns **[boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** indicating if there is a preserved view available to which the user can return.
```
Comment (bszwarc) [inline docs/luigi-client-api.md]: `Returns the` (the is missing)

File: docs/luigi-client-api.md (hunk 18 above)
Comment (bszwarc) [reply docs/luigi-client-api.md]: Will be done as part of another task.

File: docs/luigi-client-api.md (hunk 19 above)
Comment (bszwarc) [reply docs/luigi-client-api.md]: Will be done as part of another task.

File: docs/luigi-client-api.md (hunk 20 above)
Comment (bszwarc) [reply docs/luigi-client-api.md]: Will be done as part of another task.

**Pull request merged**